sql/src/main/java/org/apache/druid/sql/SqlExecutionReporter.java

```diff
@@ -105,9 +111,16 @@ public void emit()
       if (bytesWritten >= 0) {
         emitter.emit(metricBuilder.build("sqlQuery/bytes", bytesWritten));
       }
+      if (planningTimeNanos >= 0) {
+        emitter.emit(metricBuilder.build(
+            "sqlQuery/planningTimeMs",
+            TimeUnit.NANOSECONDS.toMillis(planningTimeNanos)
+        ));
+      }
 
       final Map<String, Object> statsMap = new LinkedHashMap<>();
       statsMap.put("sqlQuery/time", TimeUnit.NANOSECONDS.toMillis(queryTimeNs));
+      statsMap.put("sqlQuery/planningTimeMs", TimeUnit.NANOSECONDS.toMillis(planningTimeNanos));
```

Comment:
I'm curious that if this metric is emitted in milliseconds, why do we need to hold it in nanosecond in the code?

Reply:
I guess that it's because of the better accuracy.

```
2089/1000 - 1989/1000 = 1
```

```
(2089 - 1989) / 1000 = 0
```

`0` seems better here given the minuscule difference between two values.

Reply:
Also, my thinking was that given it doesn't need any extra space, it is better to store the time in the highest resolution. Then we can emit in whatever resolution is needed by the callers and also use the highest accuracy times for any arithmetic if needed in future.

Reply:
But I don't see if there's any chance that we might change this metric from ms to ns in the future since that would cause incompatibility.

I think SQL planning phase should be very short, does this phase always take milliseconds to complete? I guess this phase should be done in micro-seconds, if it's true, the emitted metric is always zero in most cases.

Reply:
Yes, I didn't mean to say that we'd update the current metric. Rather that incase anyone tries to use it for a new metric or derive a metric using the planningTime, they would have the value with highest accuracy.

In my local tests with `SELECT 1` query, even after multiple quick invocations the lowest planning time I saw was `4ms`. The lowest `query/time` observed was `6ms`. Have you seen < millisecond scale planning in production clusters maybe - which my local test would be missing?

Reply:
FWIW, I have seen many queries taking significant planning time. For E.g. we have seen this in particular with queries with a large IN clause or queries with many UNION operators. When debugging a slow query, we often don't have direct information about the time spent in planning. We tend to infer it by looking at other metrics and then comparing it with overall execution time. Having this metric available firsthand would help in that troubleshooting. If that planning time is too high, we would either fix it through a code change or adjust the query in some way. 

So IMO there is definitely utility in having this metric.

Reply:
@rohangarg I just guess based on my experience, and don't have evidence to prove that.